src/main/groovy/org/shipkit/internal/gradle/AutoUpdateConsumerPlugin.java

```diff
@@ -0,0 +1,126 @@
+    public static final String AUTO_UPDATE_REPLACE_VERSION = "autoUpdateReplaceVersion";
+    public static final String AUTO_UPDATE_GIT_COMMIT = "autoUpdateGitCommit";
+    public static final String AUTO_UPDATE_GIT_PUSH = "autoUpdateGitPush";
+
+    public static final String SHIPKIT_NEW_VERSION = "shipkitNewVersion";
+    public static final String SHIPKIT_DEPENDENCY_FILE = "shipkitDependencyFile";
+    public static final String SHIPKIT_DEPENDENCY_PATTERN = "shipkitDependencyPattern";
+    public static final String DEPENDENCY_PATTERN_DEFAULT = "org.shipkit:shipkit:{VERSION}";
+    public static final String DEPENDENCY_FILE_DEFAULT = "build.gradle";
+
+    @Override
+    public void apply(final Project project) {
+
+        TaskMaker.execTask(project, AUTO_UPDATE_CHECKOUT_BRANCH, new Action<Exec>() {
+            @Override
+            public void execute(final Exec exec) {
+                exec.setDescription("Creates a new branch and checks it out.");
+
+                LazyConfiguration.lazyConfiguration(exec, new Runnable() {
+                    @Override
+                    public void run() {
+                        exec.commandLine("git", "checkout", "-b", getBranchName(project));
+                    }
+                });
+            }
+        });
+
+        TaskMaker.task(project, AUTO_UPDATE_REPLACE_VERSION, ReplaceVersionTask.class, new Action<ReplaceVersionTask>() {
+            @Override
+            public void execute(final ReplaceVersionTask task) {
+                task.setDescription("Replaces Shipkit's dependency version in config file.");
+                task.mustRunAfter(AUTO_UPDATE_CHECKOUT_BRANCH);
+
+                LazyConfiguration.lazyConfiguration(task, new Runnable() {
+                    @Override
+                    public void run() {
+                        task.setNewVersion(getShipkitNewVersion(project));
+                        String file = getDependencyFile(project);
+                        task.setConfigFile(project.file(file));
+                        task.setDependencyPattern(getDependencyPattern(project));
+                    }
+                });
+            }
+        });
+
+        TaskMaker.execTask(project, AUTO_UPDATE_GIT_COMMIT, new Action<Exec>() {
+            @Override
+            public void execute(final Exec exec) {
+                exec.setDescription("Commits updated config file.");
+                exec.mustRunAfter(AUTO_UPDATE_REPLACE_VERSION);
+
+                LazyConfiguration.lazyConfiguration(exec, new Runnable() {
+                    @Override
+                    public void run() {
+                        String file = getDependencyFile(project);
+                        exec.commandLine("git", "commit", "-m", "Shipkit version updated to " + getShipkitNewVersion(project), file);
+                    }
+                });
+            }
+        });
+
+        TaskMaker.execTask(project, AUTO_UPDATE_GIT_PUSH, new Action<Exec>() {
```

Comment:
Does it make sense to use existing "GitPushTask"? The only difference seems to be using '-u' parameter here, which I don't think we really need.